### PR TITLE
User should be directly able to navigate to product detail page (#1yzcv99)

### PR DESCRIPTION
**Pull request by bashu22tiwari**

### CODE REVIEW

File: src/store/modules/product/actions.ts [hunk 1]
```diff
@@ -84,11 +84,32 @@ const actions: ActionTree<ProductState, RootState> = {
...
   },
 
-  updateCurrentProduct ({ commit, state }, payload) {
+  async updateCurrentProduct ({ commit, state }, payload) {
     // search in uploadProducts that if the clicked product is already in the upload list and set it as current product
-    const currentProduct = state.uploadProducts[payload.product.sku]
-    commit(types.PRODUCT_CURRENT_UPDATED, { product: currentProduct ? currentProduct : payload.product })
-  }
+    const currentProduct = state.uploadProducts[payload.currentSku]
+    // console.log(payload.currentSku)
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:90]: Remove unwanted comments.

File: src/store/modules/product/actions.ts [hunk 2]
```diff
@@ -84,11 +84,32 @@ const actions: ActionTree<ProductState, RootState> = {
...
+    if(currentProduct!==undefined){
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:91]:
```suggestion
    if(currentProduct){
```

File: src/views/count.vue [hunk 3]
```diff
@@ -100,6 +100,10 @@
         product: "product/getCurrent"
       })
     },
+    mounted(){
+      this.fetchProduct(this.$route.params.sku)
+      console.log("mounted", this.$route)
```
Comment (ymaheshwari1) [inline src/views/count.vue:105]: Remove console log statement

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts (hunk 2 above)
Comment (bashu22tiwari) [reply src/store/modules/product/actions.ts:90]: Updated Sir.
Thank you sir for reviewing my code.

File: src/views/count.vue (hunk 3 above)
Comment (bashu22tiwari) [reply src/views/count.vue]: Removed Sir,
Thank you sir for reviewing my code.

File: src/views/count.vue [hunk 4]
```diff
@@ -100,6 +100,12 @@
         product: "product/getCurrent"
       })
     },
+    mounted(){
+      this.fetchProduct(this.$route.params.sku)
+    },
```
Comment (ymaheshwari1) [inline src/views/count.vue:105]: We don't need mounted hook here as we are using the ionViewWillEnter hook.

File: src/views/count.vue [hunk 5]
```diff
@@ -115,6 +121,13 @@
         } else {
           showToast(translate("Enter the stock count for the product"))
         }
+      },
+      fetchProduct(sku: any) {
+        const payload = {
+          currentSku: sku,
+          queryString: '*' + sku + '*'
```
Comment (ymaheshwari1) [inline src/views/count.vue:128]: Why we are passing SKU twice, we can pass a single value and add * in the actions part, WDYT?

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts (hunk 1 above)
Comment (bashu22tiwari) [reply src/store/modules/product/actions.ts]: Removed Sir,
Thank you sir for reviewing my code.

File: src/views/count.vue (hunk 4 above)
Comment (bashu22tiwari) [reply src/views/count.vue]: Removed the mounted Hook.
Thank You sir for reviewing my code.

File: src/views/count.vue (hunk 5 above)
Comment (bashu22tiwari) [reply src/views/count.vue]: Yes, Sir, I changed the code to the same. And we are adding * in the action now.
Thank You sir for reviewing my code.

**Review by ymaheshwari1 — APPROVED** [review]

**Pull request merged**